plugins/woocommerce/client/admin/client/settings-payments/onboarding/providers/woopayments/steps/business-verification/utils/index.ts

```diff
@@ -0,0 +1,226 @@
+/**
+ * External dependencies
+ */
+import { set, toPairs } from 'lodash';
+import apiFetch from '@wordpress/api-fetch';
+import { WC_ADMIN_NAMESPACE } from '@woocommerce/data';
+
+/**
+ * Internal dependencies
+ */
+import { ListItem } from '../../../components/grouped-select-control';
+import businessTypeDescriptionStrings from '../translations/descriptions';
+import {
+	Country,
+	MccsDisplayTreeItem,
+	OnboardingFields,
+	PoEligibleData,
+	PoEligibleResponse,
+	FinalizeOnboardingResponse,
+} from '../types';
+
+export const fromDotNotation = (
+	record: Record< string, unknown >
+): Record< string, unknown > =>
+	toPairs( record ).reduce( ( result, [ key, value ] ) => {
+		return value !== null ? set( result, key, value ) : result;
+	}, {} );
+
+const hasUndefinedValues = ( obj: Record< string, unknown > ): boolean =>
+	Object.values( obj ).some( ( value ) => value === undefined );
+
+export const getAvailableCountries = (
+	countries: Record< string, string >
+): Country[] =>
+	Object.entries( countries || [] )
+		.map( ( [ key, name ] ) => ( { key, name, types: [] } ) )
+		.sort( ( a, b ) => a.name.localeCompare( b.name ) );
+
+export const getBusinessTypes = ( data: Country[] ): Country[] => {
+	return (
+		( data || [] )
+			.map( ( country ) => ( {
+				...country,
+				types: country.types.map( ( type ) => ( {
+					...type,
+					description: businessTypeDescriptionStrings[ country.key ]
+						? businessTypeDescriptionStrings[ country.key ][
+								type.key
+						  ]
+						: businessTypeDescriptionStrings.generic[ type.key ],
+				} ) ),
+			} ) )
+			.sort( ( a, b ) => a.name.localeCompare( b.name ) ) || []
+	);
+};
+
+/**
```

Comment:
Nitpick: I wonder whether we should take this opportunity to extract the API request functions to a separate file (like `actions.tsx`)? Looking at it now, I'm not sure the utils file is the best place for them.

Reply:
That sounds good, I have move all the API request functions to external file in https://github.com/woocommerce/woocommerce/pull/56588/commits/ebe28001789671b793a9e97ea3c29921e903aa1e